src/gwc/src/main/java/org/geoserver/gwc/dispatch/GwcServiceDispatcherCallback.java

```diff
@@ -29,21 +36,95 @@
 public class GwcServiceDispatcherCallback extends AbstractDispatcherCallback implements
         DispatcherCallback {
 
+    private final Catalog catalog;
+
+    public GwcServiceDispatcherCallback(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
     @Override
     public Request init(Request request) {
         String context = request.getContext();
-        if (context == null || !context.startsWith("gwc/service")) {
+        if (context == null || !context.toLowerCase().contains("gwc/service")) {
```

Comment:
I believe @jodygarnett is confused by the use of toLowerCase. OGC services are for the most part case sensitive (with the exception of key values) and the old code was not making case insensitive checks. Also, I'd probably be a bit more strict and check the path contains a known workspace and not whatever path with maybe multiple components in it.